### PR TITLE
Add placeholder login page

### LINKED ISSUE #22: Login page
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Create a login page to show that something has been already done

**Describe alternatives you've considered**
Not applicable

**Additional context**
Some basic authentication using the accounts microservice would be nice

### PR DESCRIPTION
**Pull request by maksnowak**
Closes #22

Added basic pages to frontend:
- `/` - redirects to login page now, eventually it will be some kind of a landing page with buttons for logging in and creating an account
- `/login` - login page
- `/create-account` - page for creating new accounts
- `/home` - placeholder home page, right now it only shows that the login was successful

The frontend communicates with the accounts microservice (see #32).

### CODE REVIEW

File: frontend/src/components/CreateAccountForm.vue [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    let username = event.states.username.value;
+    let email = event.states.email.value;
+    let password = event.states.password.value;
+    createAccount(username, email, password, toast);
+};
+const createAccount = async (username: string, email: string, password: string, toast: any) => {
+    let passwordSalt = Array.prototype.map.call(crypto.getRandomValues(new Uint8Array(16)), x=>(('00'+x.toString(16)).slice(-2))).join('');
+    let saltedPassword = password + passwordSalt;
+    let hashBuffer = await crypto.subtle.digest('SHA-256', new TextEncoder().encode(saltedPassword));
+    let hashedPassword = Array.prototype.map.call(new Uint8Array(hashBuffer), x=>(('00'+x.toString(16)).slice(-2))).join('');
```
Comment (Duszke333) [inline frontend/src/components/CreateAccountForm.vue:21]: This is repeated in `hashPassword` in `LoginForm`

Comment (Duszke333) [reply frontend/src/components/CreateAccountForm.vue:21]: could be moved to some helper, but not necessarily

File: frontend/src/components/LoginForm.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+const onFormSubmit = async (event: FormSubmitEvent) => {
+    let username = event.states.username.value;
+    let password = event.states.password.value;
+    await login(username, password, router, toast);
+};
+const getSalt = async (username: string) => {
+    let response = await fetch(`http://localhost:2024/login/${username}`);
+    let data = await response.json();
+    return data.passwordSalt;
+};
```
Comment (Duszke333) [inline frontend/src/components/LoginForm.vue:22]: Here you don't check for the response code. If it is not 200, there won't be a "passwordSalt" field. It's ok for prototyping, but it should be changed in the future.

**Review by Duszke333 — COMMENTED** [review]
Some minor inconveniences were found, but fixes are not necessary in the prototyping stage.

File: frontend/src/components/LoginForm.vue (hunk 2 above)
Comment (maksnowak) [reply frontend/src/components/LoginForm.vue:22]: Referenced in issue #42

**Comment by maksnowak** [comment]
> Some minor inconveniences were found, but fixes are not necessary in the prototyping stage.

Thank you, I will fix them in the future (issues #41 and #42). If everything else is ok, approve this PR please.

**Review by Duszke333 — APPROVED** [review]
Everything was converted to Issues, so LGTM 🚀

**Comment by maksnowak** [comment]
merged

**Pull request merged**